src/server/pkg/obj/minio_client.go

```diff
@@ -23,16 +24,43 @@ func newMinioClient(endpoint, bucket, id, secret string, secure bool) (*minioCli
 	}, nil
 }
 
-func (c *minioClient) Writer(name string) (io.WriteCloser, error) {
+// Represents minio writer structure with pipe and the error channel
+type minioWriter struct {
+	errChan chan error
+	pipe    *io.PipeWriter
+}
+
+// Creates a new minio writer and a go routine to upload objects to minio server
+func newMinioWriter(client *minioClient, name string) *minioWriter {
 	reader, writer := io.Pipe()
-	go func(reader *io.PipeReader) {
-		_, err := c.PutObject(c.bucket, name, reader, "application/octet-stream")
+	w := &minioWriter{
+		errChan: make(chan error),
+		pipe:    writer,
+	}
+	go func() {
+		_, err := client.PutObject(client.bucket, name, reader, "application/octet-stream")
 		if err != nil {
 			reader.CloseWithError(err)
-			return
 		}
-	}(reader)
-	return writer, nil
+		w.errChan <- err
+	}()
+	return w
+}
+
+func (w *minioWriter) Write(p []byte) (int, error) {
+	return w.pipe.Write(p)
```

Comment:
Just trying to understand the change:
- My understanding is that the goal is to make sure any Read() calls block until all pending writes are finished. Is that correct?
- Is that accomplished here? It seems like this is more or less equivalent to what we had before (delegating to w.pipe.Write, instead of returning w.pipe so that users are calling Write() on it directly) but I'm not sure I understand the mechanism here.

Thanks again for your help!